core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -451,6 +451,13 @@ private boolean fieldDefinedInClass(Object result, String attrName) {
 
     protected Object unmarshalField(final UnmarshallingContext context, final Object result, Class type, Field field) {
         Converter converter = mapper.getLocalConverter(field.getDeclaringClass(), field.getName());
+        if (converter == null) {
+            if (new RobustCollectionConverter(mapper, reflectionProvider).canConvert(type)) {
```

Comment:
It's a bit awkward to recreate this for every single field being unmarshalled, should we keep an instance as a local field just for these `canConvert` calls?